pkgs/development/tools/misc/segger-jlink/default.nix

```diff
@@ -0,0 +1,118 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, makeWrapper
+, qt4
+, libudev
+, system
+, config
+, acceptLicense ? config.segger-jlink.acceptLicense or false
+}:
+
+let
+  supported = {
+    x86_64-linux = {
+      name = "x86_64";
+      sha256 = "19gh6hlpq4dm3ji9wdf9n435zimybvgkzshdww25j9bdm0dl0595";
+    };
+    i686-linux = {
+      name = "i386";
+      sha256 = "1f0iik519panf3649nxszzh61vss927k7zgr4yx1lj991ba6rjdq";
+    };
+    aarch64-linux = {
+      name = "arm64";
+      sha256 = "0gyka0w7xiphp22vkszkx8bvm46k3pnsxmpg0mzi580gpwh3ay1q";
+    };
+    armv7l-linux = {
+      name = "arm";
+      sha256 = "0ws8l85fl4slvx6d4q779f24ppdwm901kbdbksjl66zn25wz8g33";
+    };
+  };
+
+  platform = supported.${system} or (throw "unsupported platform ${system}");
+
+  throwLicense = throw ''
+    Use of the "SEGGER JLink Software and Documentation pack" requires the
+    acceptance of the following licenses:
+
+      - SEGGER Downloads Terms of Use [1]
+      - SEGGER Software Licensing [2]
+
+    You can express acceptance by setting acceptLicense to true in your
+    configuration. Note that this is not a free license so it requires allowing
+    unfree licenses as well.
+
+    configuration.nix:
+      nixpkgs.config.allowUnfree = true;
+      nixpkgs.config.segger-jlink.acceptLicense = true;
+
+    config.nix:
+      allowUnfree = true;
+      segger-jlink.acceptLicense = true;
+
+    [1]: https://www.segger.com/downloads/jlink/JLink_Linux_V720_x86_64.tgz
```

Comment:
minor nitpick, but filling in ${system} here could facilitate them having different terms for different systems
and avoid minor confusion for non-x86_64 users

Reply:
Yes, indeed. I added the url to the let, so it is only constructed once and used in the message and the fetchurl statement.